pages/[id]/index.js

```diff
@@ -1,16 +1,63 @@
-import initialPictures from "@/store";
+import { globalPictures } from "@/store";
 import PostingDetails from "@/components/PostingDetails";
 import { useRouter } from "next/router";
+import CommentForm from "@/components/CommentForm";
+import CommentList from "@/components/CommentList";
+import { useAtom } from "jotai";
 
 export default function PostDetailsPage() {
+  const [pictures, setPictures] = useAtom(globalPictures);
   const router = useRouter();
   const { id } = router.query;
 
-  const picture = initialPictures.find((picture) => {
+  // getting the requested picture
+  const selectedPicture = pictures.find((picture) => {
     return picture.id === id;
   });
 
-  if (picture) {
-    return <PostingDetails picture={picture} />;
+  // adding comment
+  function handleAddComment(comment) {
+    setPictures(
+      pictures.map((picture) => {
+        if (selectedPicture.id === picture.id) {
+          return {
+            ...picture,
+            comments: [
+              { commentId: crypto.randomUUID(), commentContent: comment },
+              ...picture.comments,
+            ],
+          };
+        }
+        return picture;
+      })
+    );
+  }
+
+  // delete comment
+  function handleDeleteComment(deleteComment) {
+    setPictures(
+      pictures.map((picture) => {
+        if (selectedPicture.id === picture.id) {
+          const filteredComments = picture.comments.filter((comment) => {
+            return comment.commentId !== deleteComment.commentId;
+          });
+          return { ...picture, comments: filteredComments };
+        }
+        return picture;
+      })
+    );
+  }
```

Comment:
We have already talked about this function. One consideration you should make is your general data structure, currently comments are part of the post dataset. Comments could not only be assigned to posts, but later to users as well:

1. all comments of a user
2. all comments of a post

In both cases datasets have to be associated with comments. 

What I am getting at is. Users, posts and comments should be separate datasets that reference each other. A post has a userId to reference users, comments have userIds to identify authors. A user references all comments and also all posts he has written and so on.

Reply:
In my opinion, every Picture has comments. And somewhen when I'm acting with users I would add the UserId to every comment so I can display from what user the comment comes.

Reply:
Take at look at this article, it helps thinking about data-structure:

https://redux.js.org/usage/structuring-reducers/normalizing-state-shape

Reply:
There they split posts from comments, from users.